#include <iostream>
using namespace std;
int main()
{
	int n;
  	cin>>n;
  	int arr[n];
  	for(int i=0; i<n; i++) {
    	cin>>arr[i];
      	if(arr[0] == 0) {
        	cout<<"Divide by zero exception";
          	return 0;
        }
    }
  	int val = 0;
  	try {
    	if(cin>>val)
      		throw("Array Index out of Bound Exception");
    } catch(const char* c) {
    	cout<<c;
    }
	return 0;
}